### PR TITLE
build(web-components): add necessary dependencies

### PR DESCRIPTION
**Pull request by samsiegart**
refs https://github.com/Agoric/agoric-sdk/issues/6294

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -1666,6 +1666,11 @@
   resolved "https://registry.yarnpkg.com/@endo/eventual-send/-/eventual-send-0.16.5.tgz#39f2e21b5331cd9e423a8440f9e74e28fea6701e"
   integrity sha512-tWmia2jv+beGf4oVOxUqcHc358PX3Gssz4qUD3BA7D6d7AW75Pk34ArWsi29E9w6aNMrnhnpX+EaRBWoj/Q1PA==
 
+"@endo/eventual-send@^0.16.6":
+  version "0.16.6"
```
Comment (mhofman) [inline yarn.lock]: Duplicate versions of eventual send is probably not right. Please dedup or revert to 0.16.5

Comment (samsiegart) [reply yarn.lock]: Fixed thanks. I blindly took the latest version not realizing the sdk was on 0.16.5

**Comment by mhofman** [comment]
While you can manually bump versions, this won't produce the effect you want. This repo doesn't automatically publish a bare package if it's version changes, but only when a release is manually cut. However every commit merged into master gets a prerelease published on npm. You may be able to use that instead.

**Comment by samsiegart** [comment]
> While you can manually bump versions, this won't produce the effect you want.

Okay just undid that. Would be nice to have the versions line up more logically with features in this package but correct in that my concern is just that the changes are available somehow.

**Comment by mhofman** [comment]
> Would be nice to have the versions line up more logically with features in this package but correct in that my concern is just that the changes are available somehow.

This might be another argument for UI packages to live in another repo with a different release schedule from the main SDK.

cc @warner

File: packages/web-components/src/dapp-wallet-bridge/DappWalletBridge.js [hunk 2]
```diff
@@ -188,11 +188,19 @@ export class DappWalletBridge extends LitElement {
   }
 
   render() {
-    const hasBridgeHref = this.bridgeHref !== null;
+    // Locate the bridge first.
+    if (this.bridgeHref === null) {
+      return html`<agoric-iframe-messenger
```
Comment (samsiegart) [inline packages/web-components/src/dapp-wallet-bridge/DappWalletBridge.js:200]: @turadg The `<agoric-iframe-messenger>` doesn't seem to play well when you dynamically change its `src`. It only checks for a loading timeout on initial render, so collating these two as before resulted in losing a valuable error state if the bridge iframe never loaded.

**Comment by turadg** [comment]
> This might be another argument for UI packages to live in another repo with a different release schedule from the main SDK.

Agreed. Added to problem list in https://github.com/Agoric/agoric-sdk/issues/6474

File: packages/web-components/src/dapp-wallet-bridge/DappWalletBridge.js (hunk 2 above)
Comment (turadg) [reply packages/web-components/src/dapp-wallet-bridge/DappWalletBridge.js:200]: good to know. I wonder if we can make `agoric-iframe-messenger` able to handle that.

**Review by turadg — APPROVED** [review]

File: packages/web-components/src/dapp-wallet-bridge/DappWalletBridge.js (hunk 2 above)
Comment (samsiegart) [reply packages/web-components/src/dapp-wallet-bridge/DappWalletBridge.js:200]: I think we could fix it to but I'd like to keep this change contained to get it in quicker

**Comment by samsiegart** [comment]
@turadg PTAL, I made some changes to the bridge component:
1. Made the locator optional by allowing the dapp to supply the `bridgeHref` directly. I discovered that brave shields seem to only break the locator, so this allows some users to not have to disable shields to use the site.
2. Added better error handling around localstorage being blocked. I discovered that firefox by default has a shield-like button that blocks cross-site cookies, so now we get a better message like:
Screen Shot 2022-10-21 at 8 43 14 AM


Tested in chrome, firefox, and brave.

**Review by turadg — APPROVED** [review]
Approving again but I urge you to copy these files into the dapp instead of sourcing them from the SDK, so we can begin decoupling dapp release from agoric-sdk releases.

**Comment by mhofman** [comment]
I will remove myself from reviewers since the package version aspect has been resolved and this PR has morphed away from that.

**Pull request merged**